### ISSUE TITLE
Complete toPlayerState

### ISSUE THREAD

**Issue by conallor** [conallor/cycle-media#4]
I've only really added stuff that I've needed up until now, or that seemed very obvious:

```javascript
    duration: player.duration,
    volume: player.volume * 100,
    position: player.currentTime,
    playbackRate: player.playbackRate,
    source: player.currentSrc,
    width: player.videoWidth,
    height: player.videoHeight,
    isFinished: player.ended,
    isMuted: player.muted,
    isPlaying: !player.paused,
    isSeeking: player.seeking,
    isLooped: player.loop
```

There are lot more required or useful properties that should be added to the state object,

http://www.w3schools.com/tags/ref_av_dom.asp

nice visualisation of some: http://www.w3.org/2010/05/video/mediaevents.html

**Comment by HighOnDrive** [comment]
Having a way to configure these options on a per usage basis might be best. The driver source could always be tweaked but when there are so many options a configuration object might help to manage the driver use. Great visualization link!

**Comment by conallor** [comment]
I agree, it crossed my mind when I was thinking of other add ons that might come to cycle (like rewinding through time or persisting all events, for the same reason I think a throttle might be a good idea for certain events

It would be possible on the driver factory level (the current makeMediaDriver is a placeholder for that), but also on a player by player level. not sure which makes sense as sometimes its hard to look past my own use cases. For sure though, I can imagine in some cases also wanting to pass the default startWith values when creating a player (think recovering from last known state)

**Comment by HighOnDrive** [comment]
Yes, time-travel and restoring snapshots are things on my mind as well. I'm working on a music player and visualizer and syncing to video in a most configurable way would be excellent. So for dynamic rendering the player manages time and the data transport operations for a given score datagraph, then delegates to the visualizer and syncs with the video.

**Comment by conallor** [comment]
Now i think further, events/#3 is redundant if we have all the cases covered in the state$.

An obvious solution is if state$ is a function that optionally accepts property selectors:

```javascript
video.state$() => same behaviour as current, for the common use case
video.state$(... props) => only emits when selected props changes
```
You could then, if you want, handle  slow emitting props such as "isPlaying" or "width" differently than props which are changing more frequently such as "duration", where your needs suffice.

For default values discussed earlier see #5
